superset/models/helpers.py

```diff
@@ -1350,6 +1350,10 @@ def values_for_column(
         if self.fetch_values_predicate:
             qry = qry.where(self.get_fetch_values_predicate(template_processor=tp))
 
+        rls_filters = self.get_sqla_row_level_filters(template_processor=tp)
+        if rls_filters:
+            qry = qry.where(and_(*rls_filters))
```

Comment:
I *think* that you don't need the `if rls_filters:` check here, since SQLAlchemy does the smart thing if `rls_filters` is an empty list:

```python
>>> str(select(text('t')).where(and_(*[])))
'SELECT t'
```

So you could simply do:

```suggestion
        qry = qry.where(and_(*rls_filters))
```